gcloud/monitoring/group.py

```
@@ -0,0 +1,502 @@
+        this group.
+
+    :type is_cluster: boolean
+    :param is_cluster:
+        If true, the members of this group are considered to be a cluster. The
+        system can perform additional analysis on groups that are clusters.
+    """
+    def __init__(self, client, group_id=None, display_name=None,
+                 parent_id=None, filter_string=None, is_cluster=False):
+        self.client = client
+        self._id = group_id
+        self.display_name = display_name
+        self.parent_id = parent_id
+        self.filter = filter_string
+        self.is_cluster = is_cluster
+
+        if group_id:
+            self._name = group_name_from_id(client.project, group_id)
+        else:
+            self._name = None
+
+    @property
+    def id(self):
+        """Returns the group ID.
+
+        :rtype: str or None
+        :returns: the ID of the group based on it's name.
+        """
+        return self._id
+
+    @property
+    def name(self):
+        """Returns the fully qualified name of the group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the group in the format
+            "projects/<project>/groups/<id>".
+        """
+        return self._name
+
+    @property
+    def parent_name(self):
+        """Returns the fully qualified name of the parent group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the parent group.
+        """
+        if not self.parent_id:
+            return None
+        return group_name_from_id(self.client.project, self.parent_id)
+
+    @property
+    def path(self):
+        """URL path to this group.
+
+        :rtype: str
+        :returns: the path based on project and group name.
+
+        :raises: :exc:`ValueError` if :attr:`name` is not specified.
+        """
+        if not self.id:
+            raise ValueError('Cannot determine path without group ID.')
```

The path does not exist for a group without an ID - this is True for a group being created. Asking for the path of such a group should throw an error.